python/cuml/dask/ensemble/randomforestclassifier.py

```diff
@@ -357,16 +375,47 @@ def predict(self, X, output_class=True, algo='auto', threshold=0.5,
             preds = self.predict_model_on_cpu(X=X,
                                               convert_dtype=convert_dtype)
         else:
-            preds = \
-                self._predict_using_fil(X, output_class=output_class,
-                                        algo=algo,
-                                        threshold=threshold,
-                                        convert_dtype=convert_dtype,
-                                        fil_sparse_format=fil_sparse_format,
-                                        delayed=delayed)
-
+            if broadcast_data:
+                preds = \
+                    self.partial_inference(
+                        X,
+                        algo=algo,
+                        convert_dtype=convert_dtype,
+                        fil_sparse_format=fil_sparse_format,
+                        delayed=delayed
+                    )
+            else:
+                preds = \
+                    self._predict_using_fil(
+                        X,
+                        output_class=output_class,
+                        algo=algo,
+                        threshold=threshold,
+                        convert_dtype=convert_dtype,
+                        fil_sparse_format=fil_sparse_format,
+                        delayed=delayed
+                    )
         return preds
 
+    def partial_inference(self, X, delayed, **kwargs):
```

Comment:
Looks reasonable. Should resolve https://github.com/rapidsai/cuml/pull/3349#discussion_r560700072